service/history/ndc/workflow_resetter.go

```diff
@@ -100,16 +101,18 @@ func NewWorkflowResetter(
 	shardContext shard.Context,
 	workflowCache wcache.Cache,
 	logger log.Logger,
+	allowResetWithPendingChildren bool,
```

Comment:
Wondering why you put this in the constructor instead of the `ResetWorkflow` method.

Reply:
No good reason. `ResetWorkflow` already takes 15 args. So, I added it here assuming WorkflowResetter object is instantiated on every request. But it turned out to be false. The replication stack (which is in global scope) embeds the resetter object which make this global as well.
I'll move this to `ResetWorkflow` method.